kernel/src/table_features/mod.rs

```diff
@@ -148,6 +162,18 @@ pub(crate) static SUPPORTED_WRITER_FEATURES: LazyLock<HashSet<WriterFeatures>> =
             ])
         });
 
+impl ToDataType for ReaderFeatures {
+    fn to_data_type() -> DataType {
+        DataType::STRING
+    }
+}
+
+impl ToDataType for WriterFeatures {
+    fn to_data_type() -> DataType {
+        DataType::STRING
+    }
+}
```

Comment:
ah nice, so are these the only changes required to get `#[derive(Schema)]` to still work for `Protocol`?

Reply:
I believe so

Reply:
nice.